go/cmd/vtworker/status.go

```diff
@@ -112,4 +114,22 @@ func initStatusHandling() {
 			httpError(w, "worker still executing", nil)
 		}
 	})
+
+	// cancel handler
+	http.HandleFunc("/cancel", func(w http.ResponseWriter, r *http.Request) {
+		currentWorkerMutex.Lock()
+		wrk := currentWorker
+		currentWorkerMutex.Unlock()
+
+		// no worker, we go to the menu
+		if wrk == nil {
+			http.Redirect(w, r, "/", http.StatusTemporaryRedirect)
+			return
+		}
+
+		// otherwise, cancel the running worker
+		currentWorkerMutex.Lock()
```

Comment:
I suppose not - a race condition in which the worker is canceled multiple times shouldn't actually be a problem.